tools/src/test/java/org/apache/kafka/trogdor/common/MiniTrogdorCluster.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.kafka.trogdor.rest.JsonRestServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * MiniTrogdorCluster sets up a local cluster of Trogdor Agents and Coordinators.
+ */
+public class MiniTrogdorCluster implements AutoCloseable {
+    private static final Logger log = LoggerFactory.getLogger(MiniTrogdorCluster.class);
+
+    /**
+     * The MiniTrogdorCluster#Builder is used to set up a new MiniTrogdorCluster.
+     */
+    public static class Builder {
+        private final TreeSet<String> agentNames = new TreeSet<>();
+
+        private String coordinatorName = null;
+
+        private Time time = Time.SYSTEM;
+
+        private BasicPlatform.CommandRunner commandRunner =
+                new BasicPlatform.ShellCommandRunner();
+
+        private int nodeIndex = 0;
+
+        private static class NodeData {
+            String hostname;
+            AgentRestResource agentRestResource = null;
+            JsonRestServer agentRestServer = null;
+            int agentPort = 0;
+
+            JsonRestServer coordinatorRestServer = null;
+            int coordinatorPort = 0;
+            CoordinatorRestResource coordinatorRestResource = null;
+
+            Platform platform = null;
+            Agent agent = null;
+            Coordinator coordinator = null;
+
+            BasicNode node = null;
+        }
+
+        public Builder() {
+        }
+
+        /**
+         * Set the timekeeper used by this MiniTrogdorCluster.
+         */
+        public Builder time(Time time) {
+            this.time = time;
+            return this;
+        }
+
+        public Builder commandRunner(BasicPlatform.CommandRunner commandRunner) {
+            this.commandRunner = commandRunner;
+            return this;
+        }
+
+        /**
+         * Add a new trogdor coordinator node to the cluster.
+         */
+        public Builder addCoordinator(String nodeName) {
+            if (coordinatorName != null) {
+                throw new RuntimeException("At most one coordinator is allowed.");
+            }
+            coordinatorName = nodeName;
+            return this;
+        }
+
+        /**
+         * Add a new trogdor agent node to the cluster.
+         */
+        public Builder addAgent(String nodeName) {
+            if (agentNames.contains(nodeName)) {
+                throw new RuntimeException("There is already an agent on node " + nodeName);
+            }
+            agentNames.add(nodeName);
+            return this;
+        }
+
+        private NodeData getOrCreate(String nodeName, TreeMap<String, NodeData> nodes) {
+            NodeData data = nodes.get(nodeName);
+            if (data != null)
+                return data;
+            data = new NodeData();
+            data.hostname = String.format("127.0.0.%d", nodeIndex + 1);
```

Comment:
Interestingly enough, it seems that this is an OS-specific difference... On Linux, all of the 127.x.y.z addresses map to loopback.  On Mac, it looks like only 127.0.0.1 is implemented.  I have to think about how to make this work on Mac.